File: app/src/main/java/com/quranapp/android/adapters/ADPQuickReference.java
package com.quranapp.android.adapters;

import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.quranapp.android.activities.ReaderPossessingActivity;
import com.quranapp.android.components.quran.subcomponents.Verse;
import com.quranapp.android.views.reader.VerseView;

import java.util.List;

public class ADPQuickReference extends RecyclerView.Adapter<ADPQuickReference.VHReader> {
    private final ReaderPossessingActivity mActivity;
    private List<Verse> mVerses;

    public ADPQuickReference(ReaderPossessingActivity activity) {
        mActivity = activity;

        setHasStableIds(true);
    }

    public void setVerses(List<Verse> verses) {
        mVerses = verses;
    }

    @Override
    public int getItemCount() {
        return mVerses.size();
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    @NonNull
    @Override
    public VHReader onCreateViewHolder(@NonNull ViewGroup parent, int position) {
        return new VHReader(new VerseView(mActivity, parent, null, true));
    }

    @Override
    public void onBindViewHolder(@NonNull VHReader holder, int position) {
        holder.bind(mVerses.get(position));
    }


    public static class VHReader extends RecyclerView.ViewHolder {
        private final VerseView mVerseView;

        public VHReader(@NonNull VerseView verseView) {
            super(verseView);
            mVerseView = verseView;
        }

        public void bind(Verse verse) {
            if (mVerseView == null) {
                return;
            }
            mVerseView.setVerse(verse);
        }
    }
}

File: app/src/main/java/com/quranapp/android/adapters/ADPQuickReference.kt
package com.quranapp.android.adapters

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.quranapp.android.activities.ReaderPossessingActivity
import com.quranapp.android.components.quran.subcomponents.Verse
import com.quranapp.android.views.reader.VerseView

class ADPQuickReference(private val mActivity: ReaderPossessingActivity) :
    RecyclerView.Adapter<ADPQuickReference.VHReader>() {
    private var mVerses: List<Verse>? = null

    init {
        setHasStableIds(true)
    }

    fun setVerses(verses: List<Verse>?) {
        mVerses = verses
    }

    override fun getItemCount(): Int {
        return mVerses!!.size
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    override fun onCreateViewHolder(parent: ViewGroup, position: Int): VHReader {
        return VHReader(VerseView(mActivity, parent, null, true))
    }

    override fun onBindViewHolder(holder: VHReader, position: Int) {
        holder.bind(mVerses!![position])
    }

    class VHReader(verseView: VerseView) : RecyclerView.ViewHolder(verseView) {
        private val mVerseView: VerseView?

        init {
            mVerseView = verseView
        }

        fun bind(verse: Verse?) {
            if (mVerseView == null) {
                return
            }
            mVerseView.verse = verse
        }
    }
}
